pkg/version/version.go

```diff
@@ -1,29 +1,63 @@
+// Copyright © 2020 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package version
 
 import (
 	"fmt"
 	"strings"
 
 	"github.com/tektoncd/cli/pkg/cli"
+	v1 "k8s.io/api/apps/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-const pipelineNamespace = "tekton-pipelines"
-
 // GetPipelineVersion Get pipeline version, functions imported from Dashboard
 func GetPipelineVersion(c *cli.Clients) (string, error) {
 	version := ""
+	namespaces := []string{"tekton-pipelines", "openshift-pipelines"}
 
 	listOptions := metav1.ListOptions{
 		LabelSelector: "app.kubernetes.io/component=controller,app.kubernetes.io/name=tekton-pipelines",
 	}
 
-	deployments, err := c.Kube.AppsV1().Deployments(pipelineNamespace).List(listOptions)
-	if err != nil {
-		return "", err
+	var deployments []v1.Deployment
+	for _, namespace := range namespaces {
+		d, _ := c.Kube.AppsV1().Deployments(namespace).List(listOptions)
+		deployments = append(deployments, d.Items...)
+	}
+
+	if len(deployments) > 0 {
+		version = findVersion(deployments)
+	}
+
+	if version == "" {
+		d, _ := c.Kube.AppsV1().Deployments("").List(listOptions)
+		deployments = append(deployments, d.Items...)
+		version = findVersion(deployments)
+	}
+
+	if version == "" {
+		return "", fmt.Errorf("Error getting the tekton pipelines deployment version. Version is unknown")
```

Comment:
Can we get just a single word, as "unknown", reasoning is that we can script it easily... (and it would be backward compatible as before

Reply:
yes this is a function called by versionCmd, there it prints the only version if it receives an error

Reply:
Basically this error is ignored there https://github.com/tektoncd/cli/blob/master/pkg/cmd/version/version.go#L56

Reply:
👍